docs/geozarr-eopf-specification.qmd

```diff
@@ -0,0 +1,304 @@
+
+## GeoZarr V3 for EOPF - Core Requirements
+
+### 1. CF Convention Compliance
+
+All GeoZarr V3 datasets MUST follow CF (Climate and Forecast) conventions:
+
+```json
+{
+  "standard_name": "toa_bidirectional_reflectance",
+  "grid_mapping": "spatial_ref"
+}
+```
+
+**Requirements:**
+- All data variables MUST have a `standard_name` attribute
+- All geospatial data variables MUST reference a spatial reference system via `grid_mapping`
+
+### 2. Enhanced Spatial Reference System
+
+Every GeoZarr V3 dataset MUST include a `spatial_ref` coordinate variable with comprehensive CRS information:
+
+```json
+{
+  "spatial_ref": {
+    "attrs": {
+      "crs_wkt": "PROJCS[\"WGS 84 / UTM zone 32N\"...]",
+      "spatial_ref": "PROJCS[\"WGS 84 / UTM zone 32N\"...]", 
+      "epsg": 32632,
+      "proj4": "+proj=utm +zone=32 +datum=WGS84 +units=m +no_defs",
+      "GeoTransform": "300000.0 60.0 0.0 5000040.0 0.0 -60.0"
+    }
+  }
+}
+```
+
+**Requirements:**
+- MUST include `crs_wkt` in Well-Known Text format
+- SHOULD include `epsg` code when available (preserving EOPF's `proj:epsg`)
+- SHOULD include `proj4` string for additional compatibility
+- MUST include `GeoTransform` in GDAL format for pixel-to-coordinate transformation
+
+### 3. Coordinate Arrays for All Levels
+
+All GeoZarr V3 datasets MUST include proper coordinate arrays at every resolution level:
+
+```python
+{
+  "x": {
+    "dims": ["x"],
+    "data": [300000.0, 300060.0, 300120.0, ...],
+    "attrs": {
+      "units": "m",
+      "long_name": "x coordinate of projection"
+    }
+  },
+  "y": {
+    "dims": ["y"], 
+    "data": [5000040.0, 4999980.0, 4999920.0, ...],
+    "attrs": {
+      "units": "m",
+      "long_name": "y coordinate of projection"
+    }
+  }
+}
+```
+
+**Requirements:**
+- MUST include `x` and `y` coordinate arrays in the native CRS
+- Coordinates MUST be in the same units as the CRS
+- MUST include appropriate `units` and `long_name` attributes
+
+### 4. COG-Style Multiscale Support
+
+GeoZarr V3 for EOPF implements Cloud Optimized GeoTIFF (COG) style multiscale overviews:
+
+```
+eopf_dataset.zarr/
+├── zarr.json                 # Root metadata
+├── b01/                      # Native resolution (level 0)
+│   ├── zarr.json
+│   └── chunks/
+├── 1/                        # Overview level 1 (1:2 scale)
+│   ├── b01/
+│   ├── spatial_ref/
+│   ├── x/
+│   └── y/
+├── 2/                        # Overview level 2 (1:4 scale)
+│   └── ...
+└── 3/                        # Overview level 3 (1:8 scale)
+    └── ...
+```
+
+#### Multiscale Metadata
+
+Data variables with multiscale support MUST include multiscale metadata:
+
+```json
+{
+  "multiscales": {
+    "tile_matrix_set": "WebMercatorQuad",
+    "resampling_method": "nearest",
+    "tile_matrix_limits": {
+      "0": {},
+      "1": {},
+      "2": {},
+      "3": {}
+    }
+  }
+}
+```
+
+#### Overview Level Requirements
+
+Each overview level MUST:
+- Follow COG-style /2 downsampling (1:1, 1:2, 1:4, 1:8, etc.)
```

Comment:
the /2 downsampling is not mandatory in COGs

Reply:
[This is the actual requirement](https://docs.ogc.org/is/21-026/21-026.html#_requirement_reduced_resolution_subfiles_number), right:
> A COG file SHALL contain reduced resolution subfiles each one reducing the resolution by a minimum factor of 2 and a maximum factor of 10 from the previous one.
 
I think that is the type of requirement we should translate to GeoZarrr